### PR TITLE
Adding SF in 4l channel + updating btag SF csv files

### PR DESCRIPTION
**Pull request by andrea21z**

### CODE REVIEW

**Comment by klannon** [comment]
It would be good if it were really clear where our SF are coming from, in case there are questions or they need to be updated.  What about adding README files in the `topcoffea/data/*` directories with some information about where each file comes from (e.g. Twiki link or other information about their source, plus enough information that someone could figure out which file on the Twiki had been used and what it might need to be replaced with if there were an update)?  For these b-tag SFs, we could put that info in `topcoffea/data/btagSF/UL/README.md`, for example.  I'm also open to alternative approaches that accomplish the same thing.  We just want to make sure what we're doing is well enough documented that it can be understood and possibly updated later if need be.

File: analysis/topEFT/topeft.py
Comment (bryates) [inline analysis/topEFT/topeft.py:565]: For readability, it might be nice to define 2lss, 3l, and 4l variables which contain these lists. That way we can do `if ch in [2lss, 4l]: continue` to make it more clear to others reading the code.

File: analysis/topEFT/topeft.py
Comment (bryates) [inline analysis/topEFT/topeft.py:587]: As stated above, the comment above about `2lss` and `4l` categories isn't necessary with any of these, since it is clear these are electron only (or muon for the following lines)

File: topcoffea/modules/corrections.py
Comment (bryates) [inline topcoffea/modules/corrections.py:78]: What are `type3` and `type4`? It looks like below that `type4` is the lepton flavor. Should we pick better names?

**Review by bryates — CHANGES_REQUESTED** [review]
I left a few comments about some variable names, but otherwise things look good. I agree with Kevin's comment about writing where the CSV files come from, either in the README, or somewhere where they're used in the actual code.

File: topcoffea/data/fromTTH/lepSF/README.md
Comment (bryates) [inline topcoffea/data/fromTTH/lepSF/README.md:1]: Looks like this one is a copy of the main README. The other for the b-tag UL SFs looks good, thanks!

File: analysis/topEFT/topeft.py
Comment (bryates) [inline analysis/topEFT/topeft.py:164]: We should add a comment here, or maybe in the b-tag README, what working point this is. It looks like maybe `loose`.

File: topcoffea/modules/corrections.py
Comment (andrea21z) [reply topcoffea/modules/corrections.py]: Yes, type=flavor, I have updated the name

File: analysis/topEFT/topeft.py
Comment (andrea21z) [reply analysis/topEFT/topeft.py]: Those are the cuts for the UL SFs (2017, 2018) of the WP medium, I have added a comment

**Review by bryates — APPROVED** [review]

**Comment by bryates** [comment]
Thanks for all the updates @andrea21z! I think @kmohrman is going to take a look at this branch in work queue (along with the new b-tag tools), and then we can close and merge this PR.

**Comment by kmohrman** [comment]
Thanks for all of these updates @andrea21z! I tried out the code, and I think the results are expected:
* It seems to run fine locally.
* It crashes [here](https://github.com/TopEFT/topcoffea/blob/addingSF/topcoffea/modules/corrections.py#L187) when running with WQ (As expected. This happens on the master branch too, and it seems to be due to it's still using the old version of the btag stuff).
* After I locally applied the updates to `BTagScaleFactor` (same as what's currently in the `btag-sf-update` branch), the code avoids the crash mentioned above, though the code then proceeds to crash [here](https://github.com/TopEFT/topcoffea/blob/addingSF/topcoffea/modules/corrections.py#L209) when running with WQ (with the same gzip issue as was happening for the btag stuff prior to updating). I think this is also expected at this point (at least it's not unique to this branch anyway). To fix this I think we need to either find an updated way of handling the JEC stuff, or figure out why the code is trying to open the plain text files as gzip files when running with WQ.

Since those issues are not unique to this branch, I think we probably don't need to wait to merge this PR till after we figure them out (since this PR already fixes the issue of the outdated SF files, and it would be good to get those changes into the master as soon as possible).

I think the only other comment I have is about the old csv files. I think it would make sense to remove them. If we leave them in the repository, I think it might cause confusion later on, so unless I'm missing some reason why it would be good to keep them around, I think they should be removed before we merge the PR.

**Comment by bryates** [comment]
I just rebased with the master branch. The only issue was with these lines
https://github.com/TopEFT/topcoffea/blob/db2fc1a52e0ac7b879035024efa872791fc822a8/analysis/topEFT/topeft.py#L206-L209
which I picked the master versions of. Everything still runs fine.

**Pull request merged**
